cmd/cli/cli.go

```diff
@@ -211,7 +228,7 @@ func sourceWriteFunc(t targetiface.Target, ft failureiface.Failure, tr transform
 		messagesToSend := transformed.Result
 
 		res, err := retry.ExponentialWithInterface(5, time.Second, "target.Write", func() (interface{}, error) {
-			res, err := t.Write(messagesToSend)
+			res, err := t.Write(messagesToSend, btr)
```

Comment:
I started with a design that segmented them, but it left things very messy because the target needs to be aware of the transformation in order to decide how to send the data.

Similarly, the dynamic headers feature leaves us with a challenge here. Necessarily it must group data by headers, _before_ a request template is created.

From a configuration perspective, if this logic is upstream of the target, it seems very easy to break the target by configuring a separate feature.

I don't know if it's the best design, but it's what I came up with as an attempt to reconcile this with the concept of solving for batch transformations more generically.